### PR TITLE
Catchup ~1 year of wiki updates

### PR DESCRIPTION
**Pull request by Tbaut**
- WIP but review can start,  the following remains to be done:
```
INFO:	eth_getFilterLogs has a request example and only text description for response
INFO:	eth_getLogs has a request example and only text description for response

``` 
this PR
- Updates all the jsonrpc except shh (commented out for now as this is a big mess)

### CODE REVIEW

**Comment by Tbaut** [comment]
@tomusdrw can you help me for the description of following RPCs:
- parity_call 
- parity_getBlockReceipts
- parity_submitWorkDetail
- trace_callMany

File: scripts/build-rpc-markdown.js [hunk 1]
```diff
@@ -325,4 +330,4 @@ Object.keys(interfaces).sort().forEach((group) => {
   const mdFile = path.join(ROOT_DIR, `${group}.md`);
 
   fs.writeFileSync(mdFile, `${preamble}\n\n${markdown}`, 'utf8');
-});
+});
```
Comment (5chdn) [inline scripts/build-rpc-markdown.js]: nit: missing empty line

**Review by 5chdn — APPROVED** [review]
Good work!

File: src/index.js [hunk 2]
```diff
@@ -14,25 +14,23 @@
...
 module.exports = {
-  db,
   eth,
   parity,
   net,
   personal,
-  private: privateMod,
-  shh,
+  private,
+//  shh,
```
Comment (tomusdrw) [inline src/index.js]: Please add a comment or TODO here to remove that in the future

File: src/interfaces/eth.js [hunk 3]
```diff
@@ -18,6 +18,211 @@ const { Address, BlockNumber, Data, Hash, Quantity, CallRequest, TransactionRequ
...
+            type: Hash,
+            desc: '32 Bytes - hash of the block. `null` when its pending block'
+          },
+          parentHash: {
+            type: Hash,
+            desc: '32 Bytes - hash of the parent block'
+          },
+          nonce: {
+            type: Data,
+            desc: '8 Bytes - hash of the generated proof-of-work. `null` when its pending block'
```
Comment (tomusdrw) [inline src/interfaces/eth.js]: missing in case of PoA

File: src/interfaces/eth.js [hunk 4]
```diff
@@ -18,6 +18,211 @@ const { Address, BlockNumber, Data, Hash, Quantity, CallRequest, TransactionRequ
...
+    },
+    input: {
+      type: Data,
+      desc: 'the data send along with the transaction.'
+    },
+    v: {
+      type: Quantity,
+      desc: 'the standardised V field of the signature.'
+    },
+    standard_v: {
```
Comment (tomusdrw) [inline src/interfaces/eth.js]: isn't the field camel-cased?

File: src/interfaces/parity.js [hunk 5]
```diff
@@ -2043,7 +1878,35 @@ module.exports = {
...
+  lockedHardwareAccountsInfo: {
+    desc: 'Provides a list of paths to locked hardware wallets',
+    params: [],
+    returns: {
+      type: Array,
+      desc: 'Paths of all locked hardware wallets',
+      example: "['/dev/hidraw0']"
+    }
+  },
+    hardwarePinMatrixAck: {
```
Comment (tomusdrw) [inline src/interfaces/parity.js]: indentation (maybe run the code through some linter?)

File: src/interfaces/personal.js [hunk 6]
```diff
@@ -71,6 +93,82 @@ module.exports = {
...
+          "gas": "0x7f110",
+          "gasPrice": "0x09184e72a000",
+          "input": "0x603880600c6000396000f300603880600c6000396000f3603880600c6000396000f360"
+        }
+      }
+    } 
+  },
+
+  sign: {
+    desc: 'Calculates an Ethereum specific signature with: `sign(keccak256("Ethereum Signed Message: " + len(message) + message)))`.',
```
Comment (tomusdrw) [inline src/interfaces/personal.js]: First symbol is `0x19`, no?

File: src/interfaces/personal.js [hunk 7]
```diff
@@ -71,6 +93,82 @@ module.exports = {
...
+    params: [
+      {
+        type: Data,
+        desc: 'The transaction object',
```
Comment (tomusdrw) [inline src/interfaces/personal.js]: That's more of a message than transaction object.

**Review by tomusdrw — APPROVED** [review]
Looks good, couple of minor issues.

1. `parity_call` - perform multiple `eth_call` requests on top of each other, i.e. transaction `n` will be executed on top of a pending block with all `n-1` transactions applied first. Allows to get a result of execution of dependent transactions.
2. `parity_getBlockReceipts` - get receipts from all transactions from particular block, more efficient than fetching the receipts one-by-one
3. `parity_submitWorkDetail` - Same as `eth_submitWork`, but additionally returns hash of the block.
4. `trace_callMany` - perform multiple call traces on top of the same block. i.e. transaction `n` will be executed on top of a pending block with all `n-1` transactions applied (traced) first. Allows to trace dependent transactions.

**Comment by Tbaut** [comment]
Thanks a lot @seunlanlege

File: src/interfaces/eth.js (hunk 4 above)
Comment (Tbaut) [reply src/interfaces/eth.js]: nice catch

File: src/interfaces/personal.js (hunk 6 above)
Comment (Tbaut) [reply src/interfaces/personal.js]: You know better, I took it from the wiki: https://wiki.parity.io/JSONRPC-eth-module#eth_sign can you make a suggestion?

File: scripts/build-rpc-markdown.js [hunk 8]
```diff
@@ -193,6 +193,7 @@ function buildExample (name, method) {
 
   const hasReqExample = method.params.every(hasExample);
   const hasResExample = hasExample(method.returns);
+  const isPubSubApi = method.pubsub !== undefined && method.pubsub;
```
Comment (amaury1093) [inline scripts/build-rpc-markdown.js]: This can be simplified into `!!method.pubsub`

File: src/index.js [hunk 9]
```diff
@@ -14,26 +14,27 @@
 // You should have received a copy of the GNU General Public License
 // along with Parity.  If not, see <http://www.gnu.org/licenses/>.
 
-const db = require('./interfaces/db');
 const eth = require('./interfaces/eth');
 const net = require('./interfaces/net');
 const parity = require('./interfaces/parity');
 const personal = require('./interfaces/personal');
-const privateMod = require('./interfaces/private');
+const private = require('./interfaces/private');
```
Comment (amaury1093) [inline src/index.js]: `private` is a reserved keyword in JS, that's why we called it privateMod.

File: src/index.js [hunk 10]
```diff
@@ -14,26 +14,27 @@
...
-  private: privateMod,
-  shh,
+  private,
+// TODO review all shh function in rust to match with JSON RPC functions
+//  shh,
   signer,
   trace,
-  web3
+  web3,
+  secretstore
```
Comment (amaury1093) [inline src/index.js]: we have this implicit convention in JS code to order everything alphabetically, if no other natural ordering imposes itself.

File: .gitignore [hunk 11]
```diff
@@ -3,3 +3,4 @@ docs
 node_modules
 lib
 release
+push.sh
```
Comment (amaury1093) [inline .gitignore:6]: What is push.sh?

**Review by amaury1093 — APPROVED** [review]

**Comment by Tbaut** [comment]
I've seen also an example of `debug_getBadBlock` https://github.com/paritytech/parity-ethereum/pull/9502/files is that something we should consider to document or should I add an `if` to avoid anything with the `debug` module ?

asking @tomusdrw

**Comment by tomusdrw** [comment]
@Tbaut Yeah, I tihnk it would be good to document that we support that particular method. I'm not entirely sure on what other methods from `debug_` namespace we should actually support (those seem to be geth's custom extensions), that said perhaps remix-intergration would be good to have.

So yeah, let's document that instead of hiding, but maybe with a comment to the entire namespace, that we don't have plan to support all methods there (since they are not part of the official spec), although we are happy to analyze the additions case-by-case.

**Comment by Tbaut** [comment]
@seunlanlege it does not build anymore. Fixing it now. FYI you can run `npm run build:markdown` to see the errors and info.

**Comment by tomusdrw** [comment]
:tada: :beers:

**Pull request merged**